### ISSUE TITLE
Caught OutOfRangeError. Stopping Training. FIFOQueue '_3_prefetch_queue' is closed and has insufficient elements (requested 1, current size 0)

### ISSUE THREAD

**Issue by Ashiqgiga** [tensorflow/tensorflow#18325]
- **Have I written custom code (as opposed to using a stock example script provided in TensorFlow)**: No
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**:Linux Ubuntu 16.04 (PaperSpace Quadro P4000
- **TensorFlow installed from (source or binary)**: Binary
- **TensorFlow version (use command below)**: 1.70
- **Python version**:  3.6.3
- **Bazel version (if compiling from source)**:
- **GCC/Compiler version (if compiling from source)**:
- **CUDA/cuDNN version**: Cuda-repo-Ubuntu1604-9-1-local..9.1.85-1..amd64 , cudnn-9.1-linux-x64-v7.1
- **GPU model and memory**: Quadro P4000 - 8GB
- **Exact command to reproduce**:python3 train.py --logtostderr --train_dir=training/ --pipeline_config_path=training/pipeline.config

While running the above command for "faster_rcnn_resnet101_kitti" model in the PaperSpace cloud in the machine Quadro P4000 it throws the error as follows:
    
    INFO:tensorflow:Caught OutOfRangeError. Stopping Training. FIFOQueue '_3_prefetch_queue' is 
    closed and has insufficient elements (requested 1, current size 0)
	 [[Node: prefetch_queue_Dequeue = QueueDequeueV2[component_types=[DT_STRING, DT_INT32, 
    DT_FLOAT, DT_INT32, DT_FLOAT, ..., DT_INT32, DT_INT32, DT_INT32, DT_STRING, DT_INT32], 
    timeout_ms=-1, _device="/job:localhost/replica:0/task:0/device:CPU:0"](prefetch_queue)]]
	 [[Node: BatchMultiClassNonMaxSuppression/map/TensorArray_9/_2883 = 
    _Recv[client_terminated=false, recv_device="/job:localhost/replica:0/task:0/device:GPU:0", 
    send_device="/job:localhost/replica:0/task:0/device:CPU:0", send_device_incarnation=1, 
    tensor_name="edge_3507_BatchMultiClassNonMaxSuppression/map/TensorArray_9", 
    tensor_type=DT_FLOAT, _device="/job:localhost/replica:0/task:0/device:GPU:0"]()]]

    Caused by op 'prefetch_queue_Dequeue', defined at:
      File "train.py", line 167, in <module>
       tf.app.run()
      File "/home/paperspace/anaconda3/lib/python3.6/site- 
    packages/tensorflow/python/platform/app.py", line 126, in run
    _sys.exit(main(argv))
      File "train.py", line 163, in main
     worker_job_name, is_chief, FLAGS.train_dir)
     File "/home/paperspace/Desktop/models-master/research/object_detection/trainer.py", line 246, in 
    train
    clones = model_deploy.create_clones(deploy_config, model_fn, [input_queue])
    File "/home/paperspace/Desktop/models-master/research/slim/deployment/model_deploy.py", line 
    193, in create_clones
     outputs = model_fn(*args, **kwargs)
    File "/home/paperspace/Desktop/models-master/research/object_detection/trainer.py", line 158, in 
    _create_losses
     train_config.merge_multiple_label_boxes)
     File "/home/paperspace/Desktop/models-master/research/object_detection/trainer.py", line 114, in 
     get_inputs
     read_data_list = input_queue.dequeue()
     File "/home/paperspace/Desktop/models-master/research/object_detection/core/batcher.py", line 
     111, in dequeue
     batched_tensors = self._queue.dequeue()
     File "/home/paperspace/anaconda3/lib/python3.6/site- 
     packages/tensorflow/python/ops/data_flow_ops.py", line 440, in dequeue
     self._queue_ref, self._dtypes, name=name)
     File "/home/paperspace/anaconda3/lib/python3.6/site- 
     packages/tensorflow/python/ops/gen_data_flow_ops.py", line 3730, in queue_dequeue_v2
     timeout_ms=timeout_ms, name=name)
     File "/home/paperspace/anaconda3/lib/python3.6/site- 
     packages/tensorflow/python/framework/op_def_library.py", line 787, in _apply_op_helper
     op_def=op_def)
     File "/home/paperspace/anaconda3/lib/python3.6/site- 
     packages/tensorflow/python/framework/ops.py", line 3290, in create_op
     op_def=op_def)
     File "/home/paperspace/anaconda3/lib/python3.6/site- 
     packages/tensorflow/python/framework/ops.py", line 1654, in __init__
     self._traceback = self._graph._extract_stack()  # pylint: disable=protected-access

     OutOfRangeError (see above for traceback): FIFOQueue '_3_prefetch_queue' is closed and has 
     insufficient elements (requested 1, current size 0)
	 [[Node: prefetch_queue_Dequeue = QueueDequeueV2[component_types=[DT_STRING, DT_INT32, 
     DT_FLOAT, DT_INT32, DT_FLOAT, ..., DT_INT32, DT_INT32, DT_INT32, DT_STRING, DT_INT32], t 
     imeout_ms=-1, _device="/job:localhost/replica:0/task:0/device:CPU:0"](prefetch_queue)]]
	 [[Node: BatchMultiClassNonMaxSuppression/map/TensorArray_9/_2883 = 
     _Recv[client_terminated=false, recv_device="/job:localhost/replica:0/task:0/device:GPU:0", 
     send_device="/job:localhost/replica:0/task:0/device:CPU:0", send_device_incarnation=1, 
     tensor_name="edge_3507_BatchMultiClassNonMaxSuppression/map/TensorArray_9", 
     tensor_type=DT_FLOAT, _device="/job:localhost/replica:0/task:0/device:GPU:0"]()]]

     INFO:tensorflow:Finished training! Saving model to disk.

**Comment by humayun** [comment]
I got same error, Did you managed to resolve it?

**Comment by sudecki30** [comment]
I have the same error, if I undusterstand, it is a problem with the database. A file is corrumpted. But I use my jpeg only, and all files are correct. I don't understand how to resolve?

**Comment by sudecki30** [comment]
Two questions : Are your memories (RAM) full during the trainning? Do you work on black and white pictures?

**Comment by sherlockzoom** [comment]
same error.

**Comment by DavoFranco** [comment]
same error here using ssd_mobilenet_v2_coco.config. I have checked all the paths and everything seems correct

**Comment by DavoFranco** [comment]
ok I solved!!!The issue was on my ssd_mobilenet_v2_coco.config file in the input_path. I was following a tutorial and in that tutorial, the directory was data/train.record. This can work if your config file is in the same directory, but since I was following a tutorial the config file was in another folder. For instance, the issue was with the path so make sure your record's file has the right path.

**Comment by aescart1** [comment]
I had the same issue, it was because I'm used to name Tensorflow standard file *.tfrecord. Nevertheless, this time I used a generator that produce *.record and forgot to mention it in the config file.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @bignamehyp: It has been 16 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by rehmanzafar** [comment]
Solved, I had the same error and fixed it by creating the test.record and train.record files using the following two commands:
`python generate_tfrecord.py --csv_input=images/train_labels.csv --image_dir=images/train --output_path=train.record

python generate_tfrecord.py --csv_input=images/test_labels.csv --image_dir=images/test --output_path=test.record`

**Comment by aescart1** [comment]
yes, it's definitely a file not found issue. Solved as well on my side

Le dim. 27 mai 2018 à 12:58, Rehman Zafar <notifications@github.com> a
écrit :

**Comment by tensorflowbutler** [comment]
Nagging Assignee @bignamehyp: It has been 15 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by soveidadelgarmi** [comment]
Hi, I have the same issue while running open_pose training code on my own dataset. but my dataset is .mat file as depth image(grayscale). I load it with the scipy.io module in python. and repeat it on two other channels to convert 3 channels. 
but I receive this issue during training
Can someone help me?
thank you

**Comment by Anusmita-m** [comment]
> * **Have I written custom code (as opposed to using a stock example script provided in TensorFlow)**: No
> * **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**:Linux Ubuntu 16.04 (PaperSpace Quadro P4000
> * **TensorFlow installed from (source or binary)**: Binary
> * **TensorFlow version (use command below)**: 1.70
> * **Python version**:  3.6.3
> * **Bazel version (if compiling from source)**:
> * **GCC/Compiler version (if compiling from source)**:
> * **CUDA/cuDNN version**: Cuda-repo-Ubuntu1604-9-1-local..9.1.85-1..amd64 , cudnn-9.1-linux-x64-v7.1
> * **GPU model and memory**: Quadro P4000 - 8GB
> * **Exact command to reproduce**:python3 train.py --logtostderr --train_dir=training/ --pipeline_config_path=training/pipeline.config
> 
> While running the above command for "faster_rcnn_resnet101_kitti" model in the PaperSpace cloud in the machine Quadro P4000 it throws the error as follows:
> 
> ```
> INFO:tensorflow:Caught OutOfRangeError. Stopping Training. FIFOQueue '_3_prefetch_queue' is 
> closed and has insufficient elements (requested 1, current size 0)
>  [[Node: prefetch_queue_Dequeue = QueueDequeueV2[component_types=[DT_STRING, DT_INT32, 
> DT_FLOAT, DT_INT32, DT_FLOAT, ..., DT_INT32, DT_INT32, DT_INT32, DT_STRING, DT_INT32], 
> timeout_ms=-1, _device="/job:localhost/replica:0/task:0/device:CPU:0"](prefetch_queue)]]
>  [[Node: BatchMultiClassNonMaxSuppression/map/TensorArray_9/_2883 = 
> _Recv[client_terminated=false, recv_device="/job:localhost/replica:0/task:0/device:GPU:0", 
> send_device="/job:localhost/replica:0/task:0/device:CPU:0", send_device_incarnation=1, 
> tensor_name="edge_3507_BatchMultiClassNonMaxSuppression/map/TensorArray_9", 
> tensor_type=DT_FLOAT, _device="/job:localhost/replica:0/task:0/device:GPU:0"]()]]
> 
> Caused by op 'prefetch_queue_Dequeue', defined at:
>   File "train.py", line 167, in <module>
>    tf.app.run()
>   File "/home/paperspace/anaconda3/lib/python3.6/site- 
> packages/tensorflow/python/platform/app.py", line 126, in run
> _sys.exit(main(argv))
>   File "train.py", line 163, in main
>  worker_job_name, is_chief, FLAGS.train_dir)
>  File "/home/paperspace/Desktop/models-master/research/object_detection/trainer.py", line 246, in 
> train
> clones = model_deploy.create_clones(deploy_config, model_fn, [input_queue])
> File "/home/paperspace/Desktop/models-master/research/slim/deployment/model_deploy.py", line 
> 193, in create_clones
>  outputs = model_fn(*args, **kwargs)
> File "/home/paperspace/Desktop/models-master/research/object_detection/trainer.py", line 158, in 
> _create_losses
>  train_config.merge_multiple_label_boxes)
>  File "/home/paperspace/Desktop/models-master/research/object_detection/trainer.py", line 114, in 
>  get_inputs
>  read_data_list = input_queue.dequeue()
>  File "/home/paperspace/Desktop/models-master/research/object_detection/core/batcher.py", line 
>  111, in dequeue
>  batched_tensors = self._queue.dequeue()
>  File "/home/paperspace/anaconda3/lib/python3.6/site- 
>  packages/tensorflow/python/ops/data_flow_ops.py", line 440, in dequeue
>  self._queue_ref, self._dtypes, name=name)
>  File "/home/paperspace/anaconda3/lib/python3.6/site- 
>  packages/tensorflow/python/ops/gen_data_flow_ops.py", line 3730, in queue_dequeue_v2
>  timeout_ms=timeout_ms, name=name)
>  File "/home/paperspace/anaconda3/lib/python3.6/site- 
>  packages/tensorflow/python/framework/op_def_library.py", line 787, in _apply_op_helper
>  op_def=op_def)
>  File "/home/paperspace/anaconda3/lib/python3.6/site- 
>  packages/tensorflow/python/framework/ops.py", line 3290, in create_op
>  op_def=op_def)
>  File "/home/paperspace/anaconda3/lib/python3.6/site- 
>  packages/tensorflow/python/framework/ops.py", line 1654, in __init__
>  self._traceback = self._graph._extract_stack()  # pylint: disable=protected-access
> 
>  OutOfRangeError (see above for traceback): FIFOQueue '_3_prefetch_queue' is closed and has 
>  insufficient elements (requested 1, current size 0)
>  [[Node: prefetch_queue_Dequeue = QueueDequeueV2[component_types=[DT_STRING, DT_INT32, 
>  DT_FLOAT, DT_INT32, DT_FLOAT, ..., DT_INT32, DT_INT32, DT_INT32, DT_STRING, DT_INT32], t 
>  imeout_ms=-1, _device="/job:localhost/replica:0/task:0/device:CPU:0"](prefetch_queue)]]
>  [[Node: BatchMultiClassNonMaxSuppression/map/TensorArray_9/_2883 = 
>  _Recv[client_terminated=false, recv_device="/job:localhost/replica:0/task:0/device:GPU:0", 
>  send_device="/job:localhost/replica:0/task:0/device:CPU:0", send_device_incarnation=1, 
>  tensor_name="edge_3507_BatchMultiClassNonMaxSuppression/map/TensorArray_9", 
>  tensor_type=DT_FLOAT, _device="/job:localhost/replica:0/task:0/device:GPU:0"]()]]
> 
>  INFO:tensorflow:Finished training! Saving model to disk.
> ```

I changed the function for distributed training from tf.distribute.Server() to tf.train.Server() in the train.py file and it worked.
